### PR TITLE
CFG improvements

### PR DESCRIPTION
**Pull request by milenaaluisa**
Improved the CFG to handle the following statements:

`continue`: for both "while" and "doWhile" types of loop, this statement is connected to the loop condition. In the case of a "for" loop, the statement is connected to the loop step statement.

`break`: refers to a loop or switch exit. For both cases, this statement is connected to the statement that follows the loop/switch.

`switch`: this statement is connected to the first non-default case to be tested. If the switch only has one case statement, and it is the default case, then the switch is connected to this default case.

`case`: this statement is connected to the first instruction to be executed, using a true egde or an unconditional edge if it is the default case. Furthermore, if the case statement is not default, it is also connected to the next case to be tested using a false edge.

### CODE REVIEW

File: ClavaLaraApi/src-lara-clava/clava/clava/graphs/cfg/CfgBuilder.js [hunk 1]
```diff
@@ -61,6 +61,11 @@ class CfgBuilder {
    */
   #nextNodes;
 
+  /**
+   * Creates a new instance of the CfgBuilder class
+   * @param {joinpoint} $jp 
+   * @param {boolean} deterministicIds If true, uses deterministic ids for the graph ids (e.g. id_0, id_1...). Otherwise, uses $jp.astId whenever possible
```
Comment (joaobispo) [inline ClavaLaraApi/src-lara-clava/clava/clava/graphs/cfg/CfgBuilder.js:67]: Signal in the documentation that the parameter is optional, and that has a default value. The format that is being used is:

@param {boolean} [deterministicIds = false]

**Review by joaobispo — APPROVED** [review]
